src/lib/storage/boot_requirements_strategies/legacy.rb

```diff
@@ -21,18 +21,73 @@
 # To contact SUSE LLC about this file by physical or electronic mail, you may
 # find current contact information at www.suse.com.
 
-require "yast"
 require "storage/boot_requirements_strategies/base"
-require "storage/planned_volume"
-require "storage/planned_volumes_list"
-require "storage/disk_size"
 
 module Yast
   module Storage
     module BootRequirementsStrategies
       # Strategy to calculate the boot requirements in a legacy system (x86
       # without EFI)
       class Legacy < Base
+
+        using Refinements::Disk
+
+        def needed_partitions
+          volumes = super
+          volumes << grub_volume if grub_partition_required? && grub_partition_missing?
+          if mbr_gap_required?
+            mbr_gap = disk_analyzer.mbr_gap[settings.root_device]
+            # fail if gap is too small
+            if mbr_gap < DiskSize.kiB(256)
+              raise Proposal::Error
```

Comment:
Sounds better, yes.

On the other hand, raising an exception here looks like the right thing to do right now. But maybe we should discuss later if we can do something better in general. I mean, maybe the fact that the current gap is not big enough is fixable in next steps of the proposal (there is a change that we end up moving or deleting the first partition).

Anyways, something to write down and discuss later. Probably not worth considering in the end.